### PR TITLE
Add mash and capnproto (required by mash) recipes

### PR DESCRIPTION
**Pull request by mksanger**

### CODE REVIEW

**Comment by mgcam** [comment]
Tested by creating a few sketches and also by running `mash screen` on an existing sketch with previously screened data.

File: recipes/capnproto/0.7.0/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -ex
+
+cd c++
+
+autoreconf -i
+
+LD_LIBRARY_PATH=$PREFIX/lib ./configure --prefix="$PREFIX" \
```
Comment (kjsanger) [inline recipes/capnproto/0.7.0/build.sh:9]: LD_LIBRARY_PATH should be unnecessary here.

File: recipes/mash/2.3/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -ex
+
+autoreconf -i
+
+LD_LIBRARY_PATH=$PREFIX/lib ./configure --prefix="$PREFIX" \
```
Comment (kjsanger) [inline recipes/mash/2.3/build.sh:7]: LD_LIBRARY_PATH should be unnecessary here.

File: recipes/mash/2.3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - automake
+    - libtool
+    - make
+  host:
+    - boost
+    - capnproto
+    - libz-dev
+  run:
+    - boost
+    - capnproto
```
Comment (kjsanger) [inline recipes/mash/2.3/meta.yaml:34]: Please check that this is required at runtime and if so, split the capnproto recipe into two sub-packages (one with the components required to build and one with the runtime components).

**Review by kjsanger — CHANGES_REQUESTED** [review]

**Pull request merged**